src/Flysystem/Adapter/FedoraAdapter.php

```diff
@@ -298,6 +298,12 @@ public function write($path, $contents, Config $config) {
       }
     }
 
+    $parts = explode('/', $path);
+    array_pop($parts);
+    if ($parts) {
+      $this->createDir(implode('/', $parts), $config);
+    }
```

Comment:
Instead of exploding/imploding, this should probably use an implementation of [`FileSystemInterface::dirname()`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21File%21FileSystemInterface.php/function/FileSystemInterface%3A%3Adirname/10), such as [`FileSystem::dirname()`](https://api.drupal.org/api/drupal/core!lib!Drupal!Core!File!FileSystem.php/function/FileSystem%3A%3Adirname/10). PHP's [`dirname()`](https://www.php.net/dirname) _might_ work; however, `FileSystemInterface::dirname()` indicates:

> PHP's [dirname](http://php.net/dirname)() does not properly pass streams, so this function fills that gap. It is backwards compatible with normal paths and will use PHP's [dirname](http://php.net/dirname)() as a fallback.
>
> Compatibility: normal paths and stream wrappers.

Reply:
Thanks, yes that makes more sense, silly to reinvent.

I plan to verify whether this is a real issue (seems like something that would have been discovered before now?) and come back with some repro steps to confirm.